### PR TITLE
feat(docs): scope integration

### PR DESCRIPTION
**Pull request by dai-shi**
It includes both:
- [jotai-scope](https://github.com/jotaijs/jotai-scope)
- [bunshi](https://github.com/saasquatch/bunshi) (formerly `jotai-molecules`)

Close https://github.com/jotaijs/jotai-scope/issues/3

### CODE REVIEW

**Comment by convcha** [comment]
If atom does not need to be global, can I just use bunshi?
If I want to use scoped atom, I define atom as follows.

```ts
// Global atoms
const isRichTextEditorColorPaletteModalOpen = atom(false);
const isRichTextEditorFontSelectModalOpen = atom(false);

// Scoped atoms
const richTextEditorValue = atomFamily((_editorId: string) => {
  return atom("");
});
const updateRichTextEditorValue = atomFamily((_editorId: string) => {
  return atom(null, (get, set, args) => ...);
});

// Hooks
const useOpenColorPaletteModal = ...

// Components
const RichTextEditor = ({ editorId }) => {
  const openColorPaletteModal = useOpenColorPaletteModal();
  const openFontSelectModal = useOpenFontSelectModal();
  const value = useRichTextEditorValue({ editorId });
  const updateValue = useUpdateRichTextEditorValue({ editorId });

  return (
    <div>
    ...
    </div>
  );
};

const App = () => {
  return (
    <div>
      <RichTextEditor editorId="editor-1" />
      <RichTextEditor editorId="editor-2" />
    </div>
  );
};
```

**Comment by convcha** [comment]
> If atom does not need to be global

Perhaps "global" was not the right word, but this is how I use jotai anyway.

File: docs/integrations/scope.mdx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  const countAtom = useMolecule(countMolecule)
+  const [count, setCount] = useAtom(countAtom)
+  return (
+    <div>
+      {count} <button onClick={() => setCount((c) => c + 1)}>+1</button>
+    </div>
+  )
+}
+
+const App = () => (
```
Comment (DavidJFelix) [inline docs/integrations/scope.mdx:138]: I'm currently working on improving bunshi docs and I think that maybe this [example](https://github.com/saasquatch/bunshi/issues/26#issuecomment-1751805315) helps convey how bunshi works a little better. I don't think that many users will want counters based on initial value to share state.

**Review by DavidJFelix — APPROVED** [review]

File: docs/integrations/scope.mdx (hunk 1 above)
Comment (dai-shi) [reply docs/integrations/scope.mdx:138]: Hi @DavidJFelix,

I'd admit the docs and examples have huge room for improvement. Hope you can help us. I'll merge this PR. Please feel free to open a new PR to improve the example.

**Pull request merged**